codegen/IRTS/CodegenVim.hs

```diff
@@ -23,17 +23,66 @@ import qualified Vimscript.Render          as Vim
 
 codegenVim :: CodeGenerator
 codegenVim ci = do
-  let prg = runReader (genProgram (simpleDecls ci)) HM.empty
+  let decls = simpleDecls ci
+  prg <- runReaderT (genProgram decls) HM.empty
   writeFile (outputFile ci) (pretty 200 (Vim.renderProgram prg))
 
-type Gen a = Reader (HashMap Vim.Name Vim.ScopedName) a
+type Gen a = ReaderT (HashMap Vim.Name Vim.ScopedName) IO a
```

Comment:
Can't see `IO` being used anywhere? The `MonadIO` import is also redundant.

Reply:
Fixed in https://github.com/owickstrom/idris-vimscript/pull/1/commits/5372c85789485fbd480edb778012084d25069a4f